core/App/ehsm_provider.cpp

```diff
@@ -137,55 +244,27 @@ ehsm_status_t Initialize()
         sgx_destroy_enclave(g_enclave_id);
     }
 
+    // TODO: add self-test cases for FIPS-140
+
     return rc;
 }
 
 void Finalize()
 {
     sgx_destroy_enclave(g_enclave_id);
 }
+
 ehsm_status_t CreateKey(ehsm_keyblob_t *cmk)
 {
     sgx_status_t sgxStatus = SGX_ERROR_UNEXPECTED;
     sgx_status_t ret = SGX_ERROR_UNEXPECTED;
 
-    if (cmk == NULL || cmk->metadata.origin != EH_INTERNAL_KEY) {
+    if (cmk == NULL || cmk->metadata.origin != EH_INTERNAL_KEY)
+    {
         return EH_ARGUMENTS_BAD;
     }
 
-    switch (cmk->metadata.keyspec) {
-        case EH_AES_GCM_128:
-            if (cmk->keybloblen == 0) {
-                ret = enclave_create_aes_key(g_enclave_id,
-                                         &sgxStatus,
-                                         NULL,
-                                         0,
-                                         &(cmk->keybloblen));
-            } else {
-                ret = enclave_create_aes_key(g_enclave_id,
-                                         &sgxStatus,
-                                         cmk->keyblob,
-                                         cmk->keybloblen,
-                                         NULL);
-            }
-            break;
-        case EH_RSA_3072:
-            if (cmk->keybloblen == 0)
-                ret  = enclave_create_rsa_key(g_enclave_id,
-                                         &sgxStatus,
-                                         NULL,
-                                         0,
-                                         &(cmk->keybloblen));
-            else
-                ret  = enclave_create_rsa_key(g_enclave_id,
-                                         &sgxStatus,
-                                         cmk->keyblob,
-                                         cmk->keybloblen,
-                                         NULL);
-            break;
-        default:
-            return EH_KEYSPEC_INVALID;
-    }
+    ret = enclave_create_key(g_enclave_id, &sgxStatus, cmk, cmk->keybloblen);
```

Comment:
the length shoud contains the cmk-keybloblen+sizeof(struct cmk)

Reply:
it fixed